### ISSUE TITLE
Missing cmdlet: Remove-MgGroupMember

### ISSUE THREAD

**Issue by ThePoShWolf** [microsoftgraph/msgraph-sdk-powershell#452]
While ```New-MgGroupMember``` both exists and works, there is no equivalent to remove group members.

I've got v1.1.0 installed.

To check for commands that modify group memberships:

```powershell
Get-Command *groupmember* -Module microsoft.graph*
```

Group members can be removed using ```Invoke-GraphRequest```:

```powershell
Invoke-GraphRequest -Method 'Delete' -Uri "https://graph.microsoft.com/v1.0/groups/$($mgGroup.Id)/members/$($mgUser.Id)/`$ref"
```

**Comment by peombwa** [comment]
This is indeed missing in our OpenAPI docs. @irvinesunday, we need to allow delete for `groups/{group-id}/members/{member-id}/$ref`.

**Comment by ThePoShWolf** [comment]
Any update on this @peombwa ? Its been so long that I completely forgot about this issue and came here intending to open an issue about ```Remove-MgGroupMember``` missing 😆

**Comment by peombwa** [comment]
@ThePoShWolf Thanks for the reminder. we are still blocked at the OData to OpenAPI translation layer on this. I've opened an issue for the appropriate team to fix this here https://github.com/microsoftgraph/microsoft-graph-devx-api/issues/567.

**Comment by bpurchell** [comment]
@peombwa Any update on this by chance? I was scratching my head trying to figure out the command to remove a member from a group and then came across this...

**Comment by Nicostar** [comment]
I am trying to migrate all of our ADAL auth/Graph API scripts to use MSAL and MSGraph Powershell SDK, but now I also am stuck with this one. 
Can we please get an update soon and if not use the remove-mggroupmember, then atleast some sort of workaround for now? :)

**Comment by bpurchell** [comment]
> Update: I can confirm the workaround posted by ThePoShWolf works just fine.
> 
> I am trying to migrate all of our ADAL auth/Graph API scripts to use MSAL and MSGraph Powershell SDK, but now I also am stuck with this one. Can we please get an update soon? :)

@Nnicostar, what was the workaround? I don't see it and would like to use this SDK but this limitation is blocking me.

**Comment by 3vi1** [comment]
I just now stumbled across this while working on a simple script.  It's kind of ridiculous that this *basic* missing function was identified more than a year ago and still has not been resolved.  :(

**Comment by MarkDordoy** [comment]
Any update one this? ADAL goes end of life in July and you still do not have a fairly standard/expected cmdlet in place. Come on Microsoft, how can we move away from a deprecated system when you are not offering a replacement. The `Invoke-MgGraphRequest` can certainly solve for this, but its not ideal.

**Comment by JakeMoe** [comment]
Azure AD Graph (and with it, the AzureAD module) dies mid-year as well.  I'm trying to update my Azure Automation scripts to Microsoft.Graph, and recording then removing group memberships is one step in the user disable process.  I'll add myself to the list of people that would like an update on this.

**Comment by ryanchrisw** [comment]
I've been banging my head against my keyboard on this one as well. We have a TON of script that I have to get migrated over before ADAL goes in of life. There is no way you guys can EOL a product without replacements. I can forgive the lack of good documentation but the lack of replacement commands are really unacceptable.

**Comment by mikedhanson** [comment]
Any updates from the MS team?

**Comment by maisarissi** [comment]
Hello everyone.

You'all are right, the issue was not resolved. A fix was deployed back in August for the OpenAPI team, but as you pointed out, we are still missing "delete operations".

Even though it seems like a super straight forward fix, the PowerShell SDK is generated based on an OpenAPI document, which is updated regularly.
Fixing it manually would cause this to be overridden in the next version.

So, I've raised a [new issue](https://github.com/microsoftgraph/microsoft-graph-devx-api/issues/888) and I'm engaging with that team and work close to them to get this fixed as soon as possible.

**Comment by ddyett** [comment]
work dependent on an openAPI dependency so moving this out for now.

**Comment by maisarissi** [comment]
Hi @ddyett .

I thought that the work to identify and fix the few issues still remaining in the OpenAPI.OData 1.0.10 were prioritized for this month (https://github.com/microsoftgraph/microsoft-graph-devx-api/issues/902). If so, why can't we have this as part of March work since the only work on the PowerShell side is to test the OpenAPI when generating the SDK?

**Comment by peombwa** [comment]
@maisarissi, the issue I highlighted in https://github.com/microsoftgraph/microsoft-graph-devx-api/issues/902#issuecomment-1062443630 will need some engineering work in the OpenAPI.NET.OData lib before we can use the expanded OpenAPI descriptions.

**Comment by wardkeller** [comment]
> Also 1+, the graph module feels very pre-release at times when there are gaps like this and it's not great that this issue has been open since 3rd Nov 2020. There's a working solution it just needs to be wrapped up into a function and added to the module. I've just started with the Microsoft.Graph modules but I'll likely be doing a lot with graph in future, I've gotta say that the user experience so far - not just this function but other parts of these modules has been less than desirable which is a shame as they've really got a lot of potential and I'm sure a big userbase too.

Completely agree with all of your comments here. It's now June 2022 and we have less than 1 month until Azure AD cmdlets are deprecated.  I also have not being able to use the workaround. When I run the following, the AAD user I am supposed to be removing from a group is removed from Azure AD all together...

Invoke-MgGraphRequest -Method Delete -Uri "https://graph.microsoft.com/v1.0/groups/$($grp.Id)/members/$($userToRemove)/$ref" 

I did see in an example somewhere that $ref is escaped, like "`$ref". Can anyone comment on this?

**Comment by ThePoShWolf** [comment]
> I did see in an example somewhere that $ref is escaped, like "`$ref". Can anyone comment on this?

@wardkeller - Since `$ref` is inside of double quotes, PowerShell will attempt to resolve it to a variable called `$ref`, which, unless you've set one, will return `$null`. If you check the docs for [removing a group member](https://docs.microsoft.com/en-us/graph/api/group-delete-members?view=graph-rest-1.0&tabs=http#http-request), you can see exactly what the request should look like, including the `$ref` portion at the end:

> If `/$ref` is not appended to the request, the user will be deleted from Azure Active Directory (Azure AD) if the appropriate permissions are used; otherwise, a `403 Forbidden` error is returned.

**Comment by wardkeller** [comment]
> > I did see in an example somewhere that $ref is escaped, like "`$ref". Can anyone comment on this?
> 
> @wardkeller - Since `$ref` is inside of double quotes, PowerShell will attempt to resolve it to a variable called `$ref`, which, unless you've set one, will return `$null`. If you check the docs for [removing a group member](https://docs.microsoft.com/en-us/graph/api/group-delete-members?view=graph-rest-1.0&tabs=http#http-request), you can see exactly what the request should look like, including the `$ref` portion at the end:
> 
> > If `/$ref` is not appended to the request, the user will be deleted from Azure Active Directory (Azure AD) if the appropriate permissions are used; otherwise, a `403 Forbidden` error is returned.

Thanks for your reply. Makes sense now about $ref being escaped. I just tested this and it worked! Thank you.

Do we know what $ref is and why it's needed at the end?

**Comment by ThePoShWolf** [comment]
@wardkeller  - Take a look at the API docs for [removing a group member](https://docs.microsoft.com/en-us/graph/api/group-delete-members?view=graph-rest-1.0&tabs=http#http-request):

> If `/$ref` is not appended to the request, the user will be deleted from Azure Active Directory (Azure AD) if the appropriate permissions are used; otherwise, a `403 Forbidden` error is returned.

I would recommend bookmarking the Graph API docs, if you haven't already. They have helped me understand the module a lot better, especially in cases like this one. And since both the docs and the module are built from the Open API spec, they _should_ cover the same things.

**Comment by craigeaw** [comment]
For anyone coming here to see that this has been resolved, it should be noted that the new cmdlet is Remove-MgGroupMemberByRef, not Remove-MgGroupMember.

**Comment by wardkeller** [comment]
@craigeaw  - Can the new cmdlet be used now?

**Comment by craigeaw** [comment]
@wardkeller Yes. I used is successfully.

**Comment by tinuabraham** [comment]
@[craigeaw] can you confirm whether you gave -DirectoryObjectId as the id of member to be deleted in the command Remove-MgGroupMemberByRef.

**Comment by mikey32230** [comment]
> > @[craigeaw] can you confirm whether you gave -DirectoryObjectId as the id of member to be deleted in the command Remove-MgGroupMemberByRef.
> 
> Yes, I used `-DirectoryObjectId` as documented here: https://docs.microsoft.com/en-us/graph/api/group-delete-members?view=graph-rest-1.0&tabs=powershell#request

I still find it ridiculous that a standard Remove-MgGroupMember cmdlet has not been provided yet.
Can anyone confirm that Remove-MgGroupMemberByRef ONLY works for removing a **single** group member at a time. According to the docs, `-DirectoryObjectId` parameter takes a string argument, so no ability to specify multiple IDs.

In order to remove multiple group members, I'm guessing we are limited to running a loop to make numerous seperate calls to Remove-MgGroupMemberByRef.
